### DISCUSSION TITLE
Cross-Framework Component Testing Issue best approach?

### DISCUSSION THREAD

**Discussion by RayLuxembourg** [cypress-io/cypress#26087] [Component Testing]
![Screenshot 2023-03-12 at 20 33 44](https://user-images.githubusercontent.com/15327989/224565049-e12624f3-768f-49e1-84c6-40f534613a3d.png)

I have a component that needs to be tested for each framework, and I want to run the same exact test for each one. The issue is that the Cypress testing framework is not located in the same folder as the component library.

My goal is to test all components without needing separate applications or development servers for each framework. I was able to easily set up Vue and React without much effort, but Angular is causing me some trouble. Mounting Angular components alone does not work, and I receive errors related to zones and other dependencies. I am not sure if this is the intended behavior.

I am using Vite as my bundler, but I am unsure how much work the bundler does for component testing.

Overall, my end goal is to test the component on all frameworks as efficiently and effortlessly as possible. I would appreciate any opinions or experiences that others have had with this issue.

**Comment by findlay-best-wishes** [comment]
It is cool.I am building cross framwork component library, can i know about cross-framework component testing approach of you?
